sycl/test/regression/event_async_exception.cpp

```diff
@@ -0,0 +1,39 @@
+// RUN: %clangxx -fsycl %s -o %t.out
+// RUN: env SYCL_DEVICE_TYPE=HOST %t.out
+
+//==-- event_async_exception.cpp - Regression test for async exceptions ----==//
```

Comment:
I'm not sure I understand what's common between tests in regression directory and why this test goes to that directory.

Reply:
Now that you mention it, I'm not sure why these tests are separated from the rest either. Moved the test to basic_tests directory, but I don't have a strong preference one way or the other. IMO, the test directory structure is a bit of a mess right now (tests for some SYCL classes are in test/basic_tests, some are in test/CLASS_NAME, some are in test/basic_tests/CLASS_NAME, and then there is this regression directory).